### PR TITLE
Speed up CI/deploy

### PR DESCRIPTION
**Pull request by thejcannon**
Using suggestions in https://github.com/facebook/docusaurus/issues/4765 which are:

- Using `swc-loader` as a webpack JS loader
  - Full disclosure, this is greek to me. I'm just being a lemming and copying Docusaurus' own configuration
- Caching in CI/deployment

### CODE REVIEW

File: .github/workflows/ci.yml [hunk 1]
```diff
@@ -24,6 +24,15 @@ jobs:
           key: ${{ runner.os }}-website-${{ hashFiles('**/yarn.lock') }}
           restore-keys: |
             ${{ runner.os }}-website-
+      - name: Cache Docusaurus Build
+        uses: actions/cache@v3
+        with:
+          path: node_modules/.cache/webpack
+          key: docusaurus-${{ github.ref_name }}
```
Comment (huonw) [inline .github/workflows/ci.yml:31]: I think this key will cause the cache to be created once and never refreshed for each ref, which is probably fine for PRs (presumably their first build and their later builds are similar enough to likely-benefit), but... won't be good for `main`.

Maybe we could do something like https://www.pantsbuild.org/2.18/docs/using-pants/using-pants-in-ci 's coarse-grained caching with a commit or a bulk `hashFiles` in there too.

**Review by huonw — COMMENTED** [review]
Thanks for trying to make improvements here.

I've kicked off a rebuild to see how much difference this makes in the 'perfect' case (fully cached/no changes).

**Comment by huonw** [comment]
Seems like it cut the build time from:

- no-cache: 17 minutes https://github.com/pantsbuild/pantsbuild.org/actions/runs/7514593312/job/20457641489?pr=110
- cache: 9.5 minutes https://github.com/pantsbuild/pantsbuild.org/actions/runs/7514593312/job/20473072785?pr=110

So a reasonable 7.5 minute saving, but not exactly getting us to "fast" CI.

File: .github/workflows/ci.yml (hunk 1 above)
Comment (thejcannon) [reply .github/workflows/ci.yml]: Ok, see latest commit.

I got this from https://github.com/actions/cache/blob/main/tips-and-workarounds.md#update-a-cache.

As I understand it the `key` will always miss, and then go to the restore key. That'll essentially be the last build on this branch (or `main` in the case of a PR). The downside is it'll eat up cache quota like a hungry hungry hippo. But meh?

> A repository can have up to 10GB of caches. Once the 10GB limit is reached, older caches will be evicted based on when the cache was last accessed. Caches that are not accessed within the last week will also be evicted.

Comment (huonw) [reply .github/workflows/ci.yml]: Ah, yeah. We might want to nuke the cache when it gets too big (along the lines of the pants docs), or else each cache entry will eventually grow to be 5GB, 10GB, 20GB, ... by itself, and we'll spend more time uploading/downloading than it saves.

**Review by kaos — APPROVED** [review]

**Pull request merged**